nixos/modules/installer/netboot/netboot.nix

```diff
@@ -141,6 +141,21 @@ with lib;
         # /etc/NIXOS tag.
         touch /etc/NIXOS
         ${config.nix.package}/bin/nix-env -p /nix/var/nix/profiles/system --set /run/current-system
+
+        # Set password for user nixos if specified on cmdline
+        # Allows using nixos-anywhere in headless environments
+        for o in $(</proc/cmdline); do
+          case "$o" in
+            live.nixos.passwordHash=*)
+              set -- $(IFS==; echo $o)
+              sed -i "s/nixos::/nixos:$2:/" /etc/shadow
```

Comment:
I don't think `/etc/shadow` exists at this point because it's only created in the activation script. You will need an activation script as mentioned here: https://github.com/Mic92/sops-nix/blob/53c853fb1a7e4f25f68805ee25c83d5de18dc699/modules/sops/default.nix#L451

Reply:
It works for the iso image - why shouldn't it work here?

Reply:
Never mind. postBootCommands is run after the activation phase.

Reply:
Stop scaring me, please 😂